tools/smithy-rs-sync/src/main.rs

```diff
@@ -0,0 +1,409 @@
+}
+
+/// Delete any current SDK files in aws-sdk-rust. Run this before copying over new files.
+fn clean_out_existing_sdk(aws_sdk_path: &Path) -> Result<()> {
+    println!("\tcleaning out previously built SDK...");
+
+    let sdk_path = format!("{}/sdk/*", aws_sdk_path.to_string_lossy());
+    let remove_sdk_command_output = Command::new("rm")
+        .arg("-rf")
+        .arg(&sdk_path)
+        .current_dir(aws_sdk_path)
+        .output()?;
+    if !remove_sdk_command_output.status.success() {
+        bail!("failed to clean out the SDK folder at {}", sdk_path);
+    }
+
+    let examples_path = format!("{}/example/*", aws_sdk_path.to_string_lossy());
+    let remove_examples_command_output = Command::new("rm")
+        .arg("-rf")
+        .arg(&examples_path)
+        .current_dir(aws_sdk_path)
+        .output()?;
+    if !remove_examples_command_output.status.success() {
+        bail!(
+            "failed to clean out the examples folder at {}",
+            examples_path
+        );
+    }
+
+    println!("\tsuccessfully cleaned out previously built SDK");
+    Ok(())
+}
+
+/// Use `cp -r` to recursively copy all files and folders from the smithy-rs build artifacts folder
+/// to the aws-sdk-rust repo folder
+fn copy_sdk(from_path: &Path, to_path: &Path) -> Result<()> {
+    println!("\tcopying built SDK...");
+
+    let copy_sdk_command_output = Command::new("cp")
+        .arg("-r")
+        .arg(&from_path)
+        .arg(&to_path)
+        .output()?;
+    if !copy_sdk_command_output.status.success() {
+        bail!(
+            "failed to copy the built SDK from {} to {}",
+            from_path.to_string_lossy(),
+            to_path.to_string_lossy()
+        );
+    }
+
+    println!("\tsuccessfully copied built SDK");
+    Ok(())
+}
+
+/// Find the last commit made to a repo
+fn find_last_commit(repo: &Repository) -> Result<Commit> {
+    let obj = repo
+        .head()
+        .context(here!())?
+        .resolve()
+        .context(here!())?
+        .peel(ObjectType::Commit)
+        .context(here!())?;
+    obj.into_commit()
+        .map_err(|_| anyhow!("couldn't find last commit"))
+}
+
+/// Create a "mirror" commit. Works by reading a smithy-rs commit and then using the info
+/// attached to it to create a commit in aws-sdk-rust.
+fn create_mirror_commit(aws_sdk_repo: &Repository, based_on_commit: &Commit) -> Result<()> {
+    println!("\tcreating mirror commit...");
+
+    let mut index = aws_sdk_repo.index().context(here!())?;
+    // The equivalent of `git add .`
+    index
+        .add_all(["."].iter(), IndexAddOption::DEFAULT, None)
+        .context(here!())?;
+    let oid = index.write_tree().context(here!())?;
+    let parent_commit = find_last_commit(&aws_sdk_repo).context(here!())?;
+    let tree = aws_sdk_repo.find_tree(oid).context(here!())?;
+
+    let _ = aws_sdk_repo
+        .commit(
+            Some("HEAD"),
+            &based_on_commit.author(),
+            &based_on_commit.committer(),
```

Comment:
nit: in Git the committer is the one who applies the commit, so this really should be the bot's name and email. Author is used to retain original attribution.

Reply:
so this is something we should discuss: I thought the clearest thing would be to have the effective commits attributed to the person who made the codegen change.

But I could definitely be convinced we should be doing something different.

Reply:
The author field is precisely to retain authorship attribution, while the committer is the one who applies the commit. At least those are the semantics that Git ascribes: https://stackoverflow.com/questions/6755824/what-is-the-difference-between-author-and-committer-in-git Having the committer be the bot is ideal for this use case, since it lets you know that the commit was done by this program.

Reply:
I have updated this to mark the bot as the committer